module/applications/activity/check-sheet.mjs

```diff
@@ -0,0 +1,70 @@
+import * as Trait from "../../documents/actor/trait.mjs";
+import ActivitySheet from "./activity-sheet.mjs";
+
+/**
+ * Sheet for the check activity.
+ */
+export default class CheckSheet extends ActivitySheet {
+
+  /** @inheritDoc */
+  static DEFAULT_OPTIONS = {
+    classes: ["check-activity"]
+  };
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  static PARTS = {
+    ...super.PARTS,
+    effect: {
+      template: "systems/dnd5e/templates/activity/check-effect.hbs",
+      templates: [
+        ...super.PARTS.effect.templates,
+        "systems/dnd5e/templates/activity/parts/check-details.hbs"
+      ]
+    }
+  };
+
+  /* -------------------------------------------- */
+  /*  Rendering                                   */
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async _prepareEffectContext(context) {
+    context = await super._prepareEffectContext(context);
+
+    context.abilityOptions = Object.entries(CONFIG.DND5E.abilities).map(([value, config]) => ({
+      value, label: config.label
+    }));
+    if ( this.item.type === "tool" ) {
+      const ability = CONFIG.DND5E.abilities[this.item.system.ability]?.label?.toLowerCase();
+      context.abilityOptions.unshift(
```

Comment:
Every skill and tool has a default ability associated with it, so perhaps this functionality should not be limited to tool items only? We could move the 'Associated Skill or Tool' field above the 'Check Ability' field in the form, so it's a more natural flow from picking a skill to having the 'Default' option appear in the ability options?

Reply:
I think the simplest solution is to allow for a blank option on 'check ability' and have it always start blank. A user will almost always have to change this drop-down anyway. The default of 'Strength' is only going to be correct 1/6th of the time. If they create a Check Activity and forget to change it, there's no much difference between it being incorrectly Strength, and being blank.

Reply:
Okay, added blank default and set it so if no ability is found at usage time, it simply won't display a button.